src/builder.rs

```diff
@@ -567,6 +572,47 @@ impl<P: fmt::Debug, V> Bundle<InProgress<P, PartiallyAuthorized>, V> {
             |_, partial| partial,
         )
     }
+    /// Appends externally computed [`Signature`]s.
+    ///
+    /// Each signature will be applied to the one input for which it is valid. An error
+    /// will be returned if the signature is not valid for any inputs, or if it is valid
+    /// for more than one input.
```

Comment:
This is O(n<sup>2</sup>) in the number of signatures, but I guess that's okay, and I see why it was done this way.

Reply:
It's only O(n^2) for untrusted inputs, in the case where every single signature is invalid. If you've generated the signatures yourself, it's O(n).